// the-blockchain-api/dist/model/Account.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccountContext = _interopRequireDefault(require("./AccountContext"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Account model module.
 * @module model/Account
 * @version 1.0.0
 */
var Account = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Account</code>.
   * @alias module:model/Account
   */
  function Account() {
    _classCallCheck(this, Account);

    Account.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Account, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Account</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Account} obj Optional instance to populate.
     * @return {module:model/Account} The populated <code>Account</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Account();

        if (data.hasOwnProperty('context')) {
          obj['context'] = _AccountContext["default"].constructFromObject(data['context']);
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], Object);
        }

        if (data.hasOwnProperty('executable')) {
          obj['executable'] = _ApiClient["default"].convertToType(data['executable'], 'Boolean');
        }

        if (data.hasOwnProperty('owner')) {
          obj['owner'] = _ApiClient["default"].convertToType(data['owner'], 'String');
        }

        if (data.hasOwnProperty('rent_epoch')) {
          obj['rent_epoch'] = _ApiClient["default"].convertToType(data['rent_epoch'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return Account;
}();
/**
 * @member {module:model/AccountContext} context
 */


Account.prototype['context'] = undefined;
/**
 * More info about the account. What are included depends on the type of account. See examples.
 * @member {Object} data
 */

Account.prototype['data'] = undefined;
/**
 * Whether or not this account is marked as executable
 * @member {Boolean} executable
 */

Account.prototype['executable'] = undefined;
/**
 * The owner of the account
 * @member {String} owner
 */

Account.prototype['owner'] = undefined;
/**
 * @member {Number} rent_epoch
 */

Account.prototype['rent_epoch'] = undefined;
var _default = Account;
exports["default"] = _default;
