// the-blockchain-api/dist/model/GetFileResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetFileResponse model module.
 * @module model/GetFileResponse
 * @version 1.0.0
 */
var GetFileResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetFileResponse</code>.
   * @alias module:model/GetFileResponse
   */
  function GetFileResponse() {
    _classCallCheck(this, GetFileResponse);

    GetFileResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetFileResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetFileResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetFileResponse} obj Optional instance to populate.
     * @return {module:model/GetFileResponse} The populated <code>GetFileResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetFileResponse();

        if (data.hasOwnProperty('uri')) {
          obj['uri'] = _ApiClient["default"].convertToType(data['uri'], 'String');
        }

        if (data.hasOwnProperty('size')) {
          obj['size'] = _ApiClient["default"].convertToType(data['size'], 'String');
        }

        if (data.hasOwnProperty('upload_time')) {
          obj['upload_time'] = _ApiClient["default"].convertToType(data['upload_time'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GetFileResponse;
}();
/**
 * @member {String} uri
 */


GetFileResponse.prototype['uri'] = undefined;
/**
 * File size in kilobytes
 * @member {String} size
 */

GetFileResponse.prototype['size'] = undefined;
/**
 * The time when the file was uploaded
 * @member {String} upload_time
 */

GetFileResponse.prototype['upload_time'] = undefined;
var _default = GetFileResponse;
exports["default"] = _default;
