// the-blockchain-api/dist/model/ListNFTsRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ListNFTsRequest model module.
 * @module model/ListNFTsRequest
 * @version 1.0.0
 */
var ListNFTsRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListNFTsRequest</code>.
   * @alias module:model/ListNFTsRequest
   * @param publicKey {String} The public key of the account whose list of owned NFTs you want to get  
   */
  function ListNFTsRequest(publicKey) {
    _classCallCheck(this, ListNFTsRequest);

    ListNFTsRequest.initialize(this, publicKey);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ListNFTsRequest, null, [{
    key: "initialize",
    value: function initialize(obj, publicKey) {
      obj['public_key'] = publicKey;
    }
    /**
     * Constructs a <code>ListNFTsRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListNFTsRequest} obj Optional instance to populate.
     * @return {module:model/ListNFTsRequest} The populated <code>ListNFTsRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListNFTsRequest();

        if (data.hasOwnProperty('public_key')) {
          obj['public_key'] = _ApiClient["default"].convertToType(data['public_key'], 'String');
        }

        if (data.hasOwnProperty('network')) {
          obj['network'] = _ApiClient["default"].convertToType(data['network'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ListNFTsRequest;
}();
/**
 * The public key of the account whose list of owned NFTs you want to get  
 * @member {String} public_key
 */


ListNFTsRequest.prototype['public_key'] = undefined;
/**
 * @member {module:model/ListNFTsRequest.NetworkEnum} network
 * @default 'devnet'
 */

ListNFTsRequest.prototype['network'] = 'devnet';
/**
 * Allowed values for the <code>network</code> property.
 * @enum {String}
 * @readonly
 */

ListNFTsRequest['NetworkEnum'] = {
  /**
   * value: "devnet"
   * @const
   */
  "devnet": "devnet",

  /**
   * value: "mainnet-beta"
   * @const
   */
  "mainnet-beta": "mainnet-beta"
};
var _default = ListNFTsRequest;
exports["default"] = _default;
